substrate/frame/system/src/mock.rs

```diff
@@ -78,6 +78,14 @@ impl OnKilledAccount<u64> for RecordKilled {
 	}
 }
 
+#[derive(Debug, TypeInfo)]
+pub struct BlockNumberProvider;
+impl Get<u64> for BlockNumberProvider {
```

Comment:
There is already a `BlockNumberProvider` trait in `sp_runtime::traits` and also some implementations of it i think. 

Reply:
We actually needed an implementation that calls out to the System pallet. Updated the naming to clarify it better.